arm/Microsoft.Authorization/policyAssignments/readme.md

```diff
@@ -37,7 +40,7 @@ To deploy resource to a Management Group, provide the `managementGroupId` as an
 }
 ```
 
-> The name of the Management Group in the deployment does not have to match the value of the `managementGroupId` in the input parameters.
+> The name of the Management Group in the deployment does not have to match the value of the `managementGroupId` in the input parameters. For example, you can trigger the initial deployment at the root management group, but the parameter file has another management group mentioned, hence the real target is the one in the parameter file.
```

Comment:
I'm almost inclined to say we should not provide this functionality as it sounds like an edge case of an edge case.

Instead we could:
Have the template always deploy to the mgmt-group ID given with the deployment, and only deploy into other scopes if e.g. a subscriptionID/resourceGroupID was provided.

I wouldn't be suprised if this confuses people more than it helps. But maybe that's just me :D

Reply:
OK so I'll cover off the comment around scope above as well..

Good call out .. here's what'll happen
- the `managementGroupId` will be optional and set to `managementGroup().name` as default, to fetch the information from the current scope. However, the parameter will still be there in case someone wants to target different scopes. (Updating the existing functionality)
- If `subscriptionId` is provided alone, then the scope will go towards the subscription (existing functionality) 
- If `subscriptionId` and `resourceGroupName` are provided, then it'll target the resource group (existing functionality)
- I'm ok to remove this piece of information if it is confusing and we'll stick to the current ones we already have. But the feature to explicitly target a managementGroup (overriding the default) can still be there. And one of the parameter files can test that as we have 2.